app/services/course/assessment/question/koditsu_question_service.rb

```diff
@@ -78,8 +79,8 @@ def build_question_object
         templates: [{
           path: koditsu_programming_language_map[@type][:filename],
           content: @metadata['submission'],
-          prefix: @metadata['prepend'],
-          suffix: koditsu_programming_language_map[@type][:language] == 'cpp' ? '' : @metadata['append']
+          prefix: truncate_google_test_framework(@metadata['prepend']),
+          suffix: truncate_google_test_framework(@metadata['append'])
```

Comment:
this truncation is actually specific for the case when Google Test framework is being used in our prepend or append (if the pattern of this Google Test framework is not found anywhere inside prepend and append, it will do nothing). I am sure that Python (and also Java) does not use Google Test framework and so should this Google Test framework be inserted inside Python/Java question, it will raise an error during evaluation.

Therefore, we do not need to worry about the case of Python and Java

Reply:
That's not the point, the code that you write should specifically target the language if it only applies to that language.

We cannot guarantee that the text will never appear in other languages.  
If one day it does appear in a different language for whatever reason, the cost to debug the issue is going to be extremely high just because the original developer doesn't put the check for cpp language.

Reply:
done